### PR TITLE
Hide empty conversations in inbox

### PR DESCRIPTION
**Pull request by chrisnojima**
When we get the inbox we get an isEmpty flag. If that's true we ignore the inbox item unless its something we created, then we keep it for that session

@keybase/react-hackers

### CODE REVIEW

File: shared/actions/chat.js [hunk 1]
```diff
@@ -635,15 +637,25 @@ function * _loadInbox (): SagaGenerator<any, any> {
...
         yield put({type: 'chat:updateInbox', payload: {conversation}})
       }
       // find it
     } else if (incoming.chatInboxFailed) {
       incoming.chatInboxFailed.response.result()
     } else if (incoming.finished) {
       yield put({type: 'chat:updateInboxComplete', payload: undefined})
+
+      // check valid selected
+      const inboxSelector = (state: TypedState, conversationIDKey) => state.chat.get('inbox')
```
Comment (chrisnojima) [inline shared/actions/chat.js]: if it was selected and is now bad lets find a good alternate

File: shared/actions/chat.js [hunk 2]
```diff
@@ -179,7 +179,7 @@ function selectConversation (conversationIDKey: ConversationIDKey, fromUser: boo
   return {type: 'chat:selectConversation', payload: {conversationIDKey, fromUser}}
 }
 
-function _inboxConversationToConversation (convo: ConversationLocal, author: ?string, following: {[key: string]: boolean}, metaData: MetaData): ?InboxState {
+function _inboxConversationToConversation (convo: ConversationLocal, author: ?string, following: {[key: string]: boolean}, metaData: MetaData, youCreatedConversationIDKey: ?ConversationIDKey): ?InboxState {
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: Why isn't this just `youCreatedConversation: boolean`?

Comment (chrisnojima) [inline shared/actions/chat.js]: i had that originally but that meant the callers needed to extract the conversation id of the convo out of convo.info etc w/ all the null checks

Comment (MarcoPolo) [inline shared/actions/chat.js]: yeah I figured that was why. It seems a little weird to have it this way though. This function just transforms data into our InboxState type, with `youCreatedConversationIDKey` you make the caller understand that if they pass the convID that equals the convID in the given `convo` then `youCreated` flag is set. 

What about having a `conversationToIDKey` function that does the null checks and stuff

**Comment by chrisnojima** [comment]
@MarcoPolo I just moved setting this property after the creation of the object. i think it makes sense to not have this converter be a passthrough for this kind of stuff, esp since its only ever called in one place

**Comment by chrisnojima** [comment]
@keybase/react-hackers @MarcoPolo bump

**Pull request merged**